core/notifications/src/history/mod.rs

```diff
@@ -105,4 +105,11 @@ impl NotificationHistory {
     ) -> Result<(Vec<StatefulNotification>, bool), NotificationHistoryError> {
         self.repo.list_for_user(user_id, first, after).await
     }
+
+    pub async fn unacknowledged_notifications_count(
```

Comment:
this name is not idiomatic ` unacknowledged_notifications_count` looks like you have cargo culted from GQL layer.

Should probably have something like `count_unacknowledged_notifications`